### PR TITLE
feat(ai): add openai sdk integration to sentry sdk internally

### PR DESCRIPTION
**Pull request by JoshFerge**
- [x] Adds OpenAI SDK integration which is actively under development. Done here in Sentry repo instead of `sentry-python` so we can rapidly iterate on the sdk without having to do releases.
- [x] Existing test coverage of `suggested_fix` should cover this, as it is simply wrapping those function calls, and the SDK is initialized during tests.
- [x] Will be putting up a draft in the `sentry-python` repo soon, once i can figure out `tox` and the rest of our testing infra there :D

As part of https://github.com/getsentry/getsentry/issues/12171. See https://www.notion.so/sentry/LLM-v0-Tech-Spec-81a1aaae56b54c0e93606edcd882cfcd?pvs=4 for spec.

What This looks like in Sentry:
Screenshot 2023-12-04 at 2 46 10 PM
Screenshot 2023-12-04 at 2 46 05 PM
Screenshot 2023-12-04 at 2 45 54 PM

Screenshot 2023-12-04 at 2 45 47 PM

### CODE REVIEW

File: src/sentry/utils/openai_sdk_integration.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    "chat_completion_input",
+                    {"messages": kwargs.get("messages")},
+                )
+            return_value = old_open_ai_resources_chat_completions_create(*args, **kwargs)
+            if getattr(return_value, "usage", None):
+                completion_output = {
+                    "created": return_value.created,
+                    "id": return_value.id,
+                }
+                if integration.capture_prompts:
```
Comment (jennmueng) [inline src/sentry/utils/openai_sdk_integration.py:50]: would it make more sense to make the output capturing a separate option than `capture_prompts`, like `capture_response`?

**Review by jennmueng — APPROVED** [review]
just a comment on the options, otherwise lgtm

File: src/sentry/utils/openai_sdk_integration.py (hunk 1 above)
Comment (JoshFerge) [reply src/sentry/utils/openai_sdk_integration.py:50]: this is a good question. It seems like to me if you have PII in the prompt, you'd certainly risk it being in the response as well. what do you think?

Comment (jennmueng) [reply src/sentry/utils/openai_sdk_integration.py:50]: Makes sense, maybe just a simple rename this option to `capture_data` or something more universal to make it clear that this triggers capturing the response too not just the prompt?

**Pull request merged**